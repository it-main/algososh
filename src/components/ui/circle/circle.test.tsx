import { render } from "@testing-library/react";
import { Circle } from "./circle";
import { ElementStates } from "../../../types/element-states";

describe("Тестирование компонента Circle", () => {
  it("без буквы рендерится корректно", () => {
    const { asFragment } = render(<Circle />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("c буквой рендерится корректно", () => {
    const { asFragment } = render(<Circle letter="12" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("c head рендерится корректно", () => {
    const { asFragment } = render(<Circle head="12" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("c реакт-элементом в head рендерится корректно", () => {
    const { asFragment } = render(<Circle head={<Circle />} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("c tail рендерится корректно", () => {
    const { asFragment } = render(<Circle tail="tail" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("c index рендерится корректно", () => {
    const { asFragment } = render(<Circle index={15} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("c isSmall===true рендерится корректно", () => {
    const { asFragment } = render(<Circle isSmall />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("c state===Default рендерится корректно", () => {
    const { asFragment } = render(<Circle state={ElementStates.Default} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("c state===Changing рендерится корректно", () => {
    const { asFragment } = render(<Circle state={ElementStates.Changing} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("c state===Modified рендерится корректно", () => {
    const { asFragment } = render(<Circle state={ElementStates.Modified} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
