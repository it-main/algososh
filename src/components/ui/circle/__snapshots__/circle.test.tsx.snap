// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование компонента Circle c head рендерится корректно 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head string"
    >
      12
    </div>
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`Тестирование компонента Circle c index рендерится корректно 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    >
      15
    </p>
    <div
      class="text text_type_input text_color_input mt-4 absolute tail60 element"
    />
  </div>
</DocumentFragment>
`;

exports[`Тестирование компонента Circle c isSmall===true рендерится корректно 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle  small default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`Тестирование компонента Circle c state===Changing рендерится корректно 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   changing"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`Тестирование компонента Circle c state===Default рендерится корректно 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`Тестирование компонента Circle c state===Modified рендерится корректно 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   modified"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`Тестирование компонента Circle c tail рендерится корректно 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 string"
    >
      tail
    </div>
  </div>
</DocumentFragment>
`;

exports[`Тестирование компонента Circle c буквой рендерится корректно 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      >
        12
      </p>
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`Тестирование компонента Circle c реакт-элементом в head рендерится корректно 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    >
      <div
        class="content "
      >
        <div
          class="text text_type_input text_color_input mb-4 absolute head element"
        />
        <div
          class="circle   default"
        >
          <p
            class="text text_type_circle text_color_input letter"
          />
        </div>
        <p
          class="text text_type_input text_color_input mt-4 absolute index"
        />
        <div
          class="text text_type_input text_color_input mt-4 absolute tail30 element"
        />
      </div>
    </div>
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`Тестирование компонента Circle без буквы рендерится корректно 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;
